convert.py

```diff
@@ -0,0 +1,213 @@
+        formatted_purpose = (self.text_wrapper.fill(first_paragraph) +
+                tail_paragraphs)
+
+        return formatted_purpose.strip()
+
+    def format_inputs(self, function):
+        def param_replacement(match):
+            return r'\param %s:' % match.group(1)
+
+        dedented = textwrap.dedent(function.inputs)
+        text, _ = re.subn(r'\n\s+', ' ', dedented, flags=re.MULTILINE)
+        text, num_replacements = re.subn(r'^([a-zA-Z0-9_]+)\s+[:-]',
+                param_replacement, text, flags=re.MULTILINE)
+
+        if num_replacements == 0:
+            text = r'parameters: %s' % text
+
+        text = '\n'.join(self.input_wrapper.fill(t) for t in text.split('\n'))
+        return text.strip()
+
+    def format_returns(self, function):
+        subbed, _ = self.whitespace_re.subn(' ', function.returns)
+        return self.input_wrapper.fill(r'\returns %s' % subbed)
+
+
+def convert_function_doc(function, file, doc_width):
+    """ Return a doxygen-style doc string for the supplied Function.  """
+    formatter = FunctionFormatter(doc_width)
+
+    sections = []
+
+    if function.purpose and function.purpose.strip():
+        sections.append(formatter.format_purpose(function))
+
+    if function.inputs and function.inputs.strip():
+        sections.append(formatter.format_inputs(function))
+
+    if function.returns and function.returns.strip():
+        sections.append(formatter.format_returns(function))
+
+    if sections:
+        text = '\n\n'.join(sections)
+        if text:
+            text = make_doxy_comment(text)
+        return text + '\n'
+
+    return ''
+
+
+def is_class_doc(block):
+    """ Return whether the block appears to be a class doc block.  """
+    return has_field(block, 'Class')
+
+
+def convert_class_doc(c, doc_width):
+    """ Return a doxygen-style class string.  """
+    text_wrapper = textwrap.TextWrapper(width=doc_width)
+    stripped = c.purpose.strip()
+    return (make_doxy_comment(text_wrapper.fill(stripped)) + '\n'
+            if stripped else '')
+
+
+def replace_block(start, block_contents, file, doc_width):
+    """
+    Replace an old-style documentation block with the doxygen equivalent
+    """
+    block = Block(start,
+            {f.name: f.contents
+                for f in parse_fields(block_contents, doc_width)})
+
+    if is_header_doc(block):
+        return convert_header_doc(header_from_block(block), doc_width)
+
+    if is_function_doc(block):
+        return convert_function_doc(
+                function_from_block(block), file, doc_width)
+
+    if is_class_doc(block):
+        return convert_class_doc(class_from_block(block), doc_width)
+
+    warn('block in "%s" has unrecognised format:\n%s' %
+            (file, block_contents))
+
+    return ''
+
+
+def convert_file(file):
+    """ Replace documentation in file with doxygen-styled comments.  """
+    with open(file) as f:
+        contents = f.read()
+
+    doc_width = 75
```

Comment:
This is a misleading name, I think - this is the width used for wrapped text, but then the leading `/// ` is added to the beginning of each line, meaning that the actual width of the block is 79 columns. This could be 76, if we want to go to 80 columns.